File: vol2/ch17/LeetCode167-2.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        n=len(numbers)
        for i in range(0,n-1):
            tmp=numbers[i+1:n]
            it=bisect.bisect_left(tmp,target-numbers[i])
            if it!=len(tmp) and numbers[i]+numbers[it+i+1]==target:
                return [i+1,it+i+2]
        return [0,0]

File: vol2/ch17/LeetCode167-2.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers,int target) {
        int n=numbers.size();
        for(int i=0;i<n-1;i++) {
            auto it=lower_bound(numbers.begin()+i+1,numbers.end(),target-numbers[i]);
            int j=it-numbers.begin();
            if(it!=numbers.end() && numbers[i]+*it==target)
                return {i+1,j+1};
        }
        return {0,0};
    }
};

